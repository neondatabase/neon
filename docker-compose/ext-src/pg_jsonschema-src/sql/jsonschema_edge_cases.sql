-- Schema with enums, nulls, extra properties disallowed
SELECT jsonschema_is_valid('{
  "type": "object",
  "properties": {
    "status": { "type": "string", "enum": ["active", "inactive", "pending"] },
    "email": { "type": ["string", "null"], "format": "email" }
  },
  "required": ["status"],
  "additionalProperties": false
}'::json);

-- Valid enum and null email
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "status": { "type": "string", "enum": ["active", "inactive", "pending"] },
      "email": { "type": ["string", "null"], "format": "email" }
    },
    "required": ["status"],
    "additionalProperties": false
  }'::json,
  '{"status": "active", "email": null}'::json
);

-- Invalid enum value
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "status": { "type": "string", "enum": ["active", "inactive", "pending"] },
      "email": { "type": ["string", "null"], "format": "email" }
    },
    "required": ["status"],
    "additionalProperties": false
  }'::json,
  '{"status": "disabled", "email": null}'::json
);

-- Invalid email format (assuming format is validated)
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "status": { "type": "string", "enum": ["active", "inactive", "pending"] },
      "email": { "type": ["string", "null"], "format": "email" }
    },
    "required": ["status"],
    "additionalProperties": false
  }'::json,
  '{"status": "active", "email": "not-an-email"}'::json
);

-- Extra property not allowed
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "status": { "type": "string", "enum": ["active", "inactive", "pending"] },
      "email": { "type": ["string", "null"], "format": "email" }
    },
    "required": ["status"],
    "additionalProperties": false
  }'::json,
  '{"status": "active", "extra": "should not be here"}'::json
);
