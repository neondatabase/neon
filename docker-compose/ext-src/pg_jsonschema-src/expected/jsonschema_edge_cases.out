-- Schema with enums, nulls, extra properties disallowed
SELECT jsonschema_is_valid('{
  "type": "object",
  "properties": {
    "status": { "type": "string", "enum": ["active", "inactive", "pending"] },
    "email": { "type": ["string", "null"], "format": "email" }
  },
  "required": ["status"],
  "additionalProperties": false
}'::json);
 jsonschema_is_valid 
---------------------
 t
(1 row)

-- Valid enum and null email
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "status": { "type": "string", "enum": ["active", "inactive", "pending"] },
      "email": { "type": ["string", "null"], "format": "email" }
    },
    "required": ["status"],
    "additionalProperties": false
  }'::json,
  '{"status": "active", "email": null}'::json
);
 jsonschema_validation_errors 
------------------------------
 {}
(1 row)

-- Invalid enum value
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "status": { "type": "string", "enum": ["active", "inactive", "pending"] },
      "email": { "type": ["string", "null"], "format": "email" }
    },
    "required": ["status"],
    "additionalProperties": false
  }'::json,
  '{"status": "disabled", "email": null}'::json
);
                     jsonschema_validation_errors                     
----------------------------------------------------------------------
 {"\"disabled\" is not one of [\"active\",\"inactive\",\"pending\"]"}
(1 row)

-- Invalid email format (assuming format is validated)
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "status": { "type": "string", "enum": ["active", "inactive", "pending"] },
      "email": { "type": ["string", "null"], "format": "email" }
    },
    "required": ["status"],
    "additionalProperties": false
  }'::json,
  '{"status": "active", "email": "not-an-email"}'::json
);
      jsonschema_validation_errors       
-----------------------------------------
 {"\"not-an-email\" is not a \"email\""}
(1 row)

-- Extra property not allowed
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "status": { "type": "string", "enum": ["active", "inactive", "pending"] },
      "email": { "type": ["string", "null"], "format": "email" }
    },
    "required": ["status"],
    "additionalProperties": false
  }'::json,
  '{"status": "active", "extra": "should not be here"}'::json
);
                    jsonschema_validation_errors                    
--------------------------------------------------------------------
 {"Additional properties are not allowed ('extra' was unexpected)"}
(1 row)

